apps/spa/src/pages/Connect4Page.tsx

```diff
@@ -1,50 +1,112 @@
-import React, { useReducer } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import GameGrid from "@components/GameGrid";
 import { useSearchParams } from "react-router-dom";
-import { playToken } from "../../../../packages/shared/lib/connect4";
-import useInitGameState from "@hooks/useInitGameState";
-import type { GameState } from "../../../../packages/shared/types/gameState";
+import {
+  PlayerNum,
+  type Game,
+} from "../../../../packages/shared/types/gameState";
+import ShareButton from "@components/ShareButton";
+import { getGame, getGameStateVersion } from "@services/api";
+import GameInfos from "@components/GameInfos";
 
-export const BOARD_STATE_QP = "state";
-export const BOARD_STATE_QP_SEPARATOR_CHAR = ",";
+export const PLAYER_NUM_QP = "playerNum";
+export const GAME_ID_QP = "gameId";
+export const SHARED_GAME_ID_QP = "sharedGameId";
+
+async function isGameStateNeedsUpdate(game: Game): Promise<boolean> {
```

Comment:
```suggestion
async function doesGameStateNeedUpdate(game: Game): Promise<boolean> {
```

Otherwise it's not English